### ISSUE TITLE
No quotation signs for non-strings

### ISSUE THREAD

**Issue by Spixmaster** [adrienverge/yamllint#814]
For example boolean values should not be quoted but only strings. Maybe add a rule to `quoted-strings`.

```yaml
foo: "false"
```

**Comment by adrienverge** [comment]
`"false"` is a valid string, it should not be reported by `quoted-strings`. Otherthise I'm not sure I understand your point, please make clear examples of what you expect / what yamllint does instead.

**Comment by Spixmaster** [comment]
My motive is to have deterministic, convergent behaviour. Therefore, I want only strings to be quoted. However, you are probably right. `false` is a valid string. I think this is the same case for `[]`, `{}`, `null` and numbers.

Is there no mechanism to prevent a user to quote values other than strings?
